design/deployment.md

```diff
@@ -0,0 +1,98 @@
+# A deployment job
+
+A "deployment" job is a collection of steps to be run sequentially against the environment. 
+We recommend specifying deployment steps in a deployment job.  
+
+## Scenarios
+
+- Run a set of steps against an environment. Record deployment history and status of the deployments.
+- Perform safe deployments. In other words, define how your application is rolled-out.
+
+  
+### Schema
+
+```yaml
+jobs:
+- deployment: string  # name of the deployment job, A-Z, a-z, 0-9, and underscore
+  displayName: string  # friendly name to display in the UI
+  dependsOn: string | [ string ]
+  environment: string # target environment name and optionally a resource-name to record the deployment history; format: <environment-name>.<resource-name>
+  condition: string
+  continueOnError: boolean  # 'true' if future jobs should run even if this job fails; defaults to 'false'
+  pool: pool # see pool schema
```

Comment:
When a pool is not specified, regular job assumes it to be hosted. For deployment job, this assumption may be incorrect. I believe we would want to use the Pool (for VM resources) in Environment implicitly instead.

Reply:
Yes, pool is optional in case of VM targets. For other resources we still rely on pool. There can be some optimizations where user can leverage an existing VM agent for remote work loads, if the deployment job is targeting more than one type of resource (VM, web apps)